src/ui/public/vislib/styles/_legend.less

```diff
@@ -54,6 +69,15 @@ visualize-legend {
     text-align: left;
 
     flex-direction: column;
+
+    .vis-container--legend-top &,
+    .vis-container--legend-bottom & {
+      width: auto;
+
+      li {
```

Comment:
Instead of using the `li` element in this selector, how about the `.legend-value` instead? This way the intention behind the selector is a little clearer.